### PR TITLE
Download envelopes lazily from toitlang/envelopes

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: cmd/jag/commands/envelope.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/toitlang/jaguar/cmd/jag/directory"
+)
+
+// getCachedFirmwareEnvelopePath returns the path to the cached firmware envelope.
```
Comment (kasperl) [inline cmd/jag/commands/envelope.go:17]:
```suggestion
// GetCachedFirmwareEnvelopePath returns the path to the cached firmware envelope.
```

File: cmd/jag/commands/envelope.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	fmt.Printf("Successfully installed %s firmware into %s\n", model, assetsPath)
+	return nil
+}
+
+func GetFirmwareEnvelopeFileName(model string) string {
+	return fmt.Sprintf("firmware-%s.envelope", model)
+}
+
+// GetFirmwareEnvelopePath returns the firmware envelope path for the given model.
```
Comment (kasperl) [inline cmd/jag/commands/envelope.go:81]:
```suggestion
// getFirmwareEnvelopePath returns the firmware envelope path for the given model.
```

File: cmd/jag/commands/envelope.go (hunk 1 above)
Comment (floitsch) [reply cmd/jag/commands/envelope.go:17]: done.

File: cmd/jag/commands/envelope.go (hunk 2 above)
Comment (floitsch) [reply cmd/jag/commands/envelope.go:81]: done.

**Comment by floitsch** [comment]
Forgot to upload the feedback changes.
Doing it in a follow-up PR.

**Pull request merged**